pkg/app/desired_state_file_loader.go

```diff
@@ -190,33 +174,40 @@ func (a *desiredStateLoader) load(yaml []byte, baseDir, file string, evaluateBas
 	return st, nil
 }
 
-func (ld *desiredStateLoader) renderAndLoad(env, overrodeEnv *environment.Environment, baseDir, filename string, content []byte, evaluateBases bool) (*state.HelmState, error) {
+func (ld *desiredStateLoader) load(env, overrodeEnv *environment.Environment, baseDir, filename string, content []byte, evaluateBases bool) (*state.HelmState, error) {
 	// Allows part-splitting to work with CLRF-ed content
 	normalizedContent := bytes.ReplaceAll(content, []byte("\r\n"), []byte("\n"))
 	parts := bytes.Split(normalizedContent, []byte("\n---\n"))
 
 	var finalState *state.HelmState
 
 	for i, part := range parts {
-		var yamlBuf *bytes.Buffer
-		var err error
-
 		id := fmt.Sprintf("%s.part.%d", filename, i)
 
-		if env == nil && overrodeEnv == nil {
-			yamlBuf, err = ld.renderTemplatesToYaml(baseDir, id, part)
-			if err != nil {
-				return nil, fmt.Errorf("error during %s parsing: %v", id, err)
+		var rawContent []byte
+
+		if filepath.Ext(filename) == ".gotmpl" || !runtime.V1Mode {
```

Comment:
@yxxhero This is my interpretation of your `isRequiredRender`. See https://github.com/helmfile/helmfile/pull/440/files#r1055911855